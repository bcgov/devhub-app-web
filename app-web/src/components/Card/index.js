/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

// common card layout components
import styled from '@emotion/styled';
import { RESOURCE_TYPES, TOPICS, SEARCH_RESOURCE_TYPES, JOURNEY } from '../../constants/ui';
import Dotdotdot from 'react-dotdotdot';
import Link from '../UI/Link/Link';
import { EventLogoWrapper, DateLI, EventDetailsContainer } from '../Event';
export { BaseCard, Card } from './Card';

export const Container = styled.article`
  width: 265px;
  height: 238.5px;
  margin: 10px 0;
  border: 1px solid #ccc;
  border-top: transparent;
  border-radius: 2px;
  box-sizing: border-box;
  transition: transform 0.25s ease-out;
  box-shadow: 0 1px 2px 1px #00000026;
  display: flex;
  flex-flow: column nowrap;
  overflow: hidden;
  :hover {
    box-shadow: 0 2px 2px 1px #00000026;
    transform: translateY(-2px);
  }
`;

export const DecorativeBar = styled.div`
  background-color: ${props => props.theme.colors[props.color]};
  flex: 0 0 10px;
  min-height: 10px;
`;

export const Title = styled(Dotdotdot)`
  font-size: 20px;
  color: ${({ theme }) => theme.colors.grey};
  line-height: 1.3;
  margin-bottom: 0;
  flex: 0 0 auto;
  text-transform: capitalize;
  word-break: break-word;
`;

export const Description = styled(Dotdotdot)`
  line-height: 1.4;
  font-size: 14px;
  flex: 0 0 auto;
  margin-bottom: 0;
  margin-top: 4px;
  text-transform: none;
`;

export const LinkWrapper = styled(Link)`
  text-decoration: none;
  color: initial;
  :hover {
    text-decoration: none;
    color: initial;
    cursor: pointer;
  }
`;

/**
 * used to predictably lay out titles/ descriptions/images in a flex box within a card
 */
export const CardBody = styled.div`
  padding: 6px 10px;
  height: 100%;
  overflow: auto;
  height: 100%;
  display: flex;
  flex: 1 1 auto;
  flex-flow: column nowrap;
`;

// combination of card image and wrapper make images response correctly
// to the flex container. The card image wrapper can grow/shrink automatically
// the object-fit scaledown maintains aspect ratio for any type of image
export const Image = styled.img`
  width: 100%;
  height: auto;
  object-fit: scale-down;
  max-height: 100%;
  margin-bottom: 0;
`;

export const ImageWrapper = styled.div`
  flex: 1 1 auto;
  align-self: center;
  overflow: hidden;
  padding-top: 10px;
  text-align: center;
  display: flex;
`;

export const EventInfoDiv = styled.div`
  > li {
    color: #494949;
    margin-bottom: 0px;
  }
`;

export const EventDate = styled(DateLI)`
  border-radius: 0px;
  border: 0px;
  margin: 0;
  > span {
    margin-top: -15px;
    color: #494949;
    font-weight: 1000;
  }
  > small {
    bottom: 0.1em;
  }
`;

export const EventContainer = styled(EventDetailsContainer)`
  height: 64px;
  font-size: 15px;
  margin-left: -10px;
  background: rgba(229, 229, 229, 0.8);
  width: 100%;
  > li {
    margin-bottom: 0px;
  }
`;

// special cases to handle presentation of the eventbrite and meetup logo inside of a card
// please note that although meetups is no longer being implemented (since the api disabled access token authenticated)
// it may be reactivated at one point, for this reason these components are being left behind
export const EventbriteImageWrapper = styled(EventLogoWrapper)`
  margin-top: 0px;
  img {
    margin-left: -8px;
    padding-left: 0px;
  }
`;

export const MeetupImageWrapper = styled(EventLogoWrapper)`
  margin: -1px 0px 5px 0px;
  img {
    padding-left: 0px;
  }
`;

export const CONTENT = {
  byResourceType: {
    [RESOURCE_TYPES.COMPONENTS]: {
      id: 'RESOURCE_TYPES.COMPONENTS',
      text: 'Component',
    },
    [RESOURCE_TYPES.DOCUMENTATION]: {
      id: 'RESOURCE_TYPES.DOCUMENTATION',
      text: 'Documentation',
    },
    [RESOURCE_TYPES.SELF_SERVICE_TOOLS]: {
      id: 'RESOURCE_TYPES.SELF_SERVICE_TOOLS',
      text: 'Tool',
    },
    [RESOURCE_TYPES.PEOPLE]: {
      id: 'RESOURCE_TYPES.PEOPLE',
      text: 'Contact',
    },
    [RESOURCE_TYPES.REPOSITORIES]: {
      id: 'RESOURCE_TYPES.REPOSITORIES',
      text: 'Repository',
    },
    [TOPICS]: {
      id: 'RESOURCE_TYPES.TOPICS',
      text: 'Topic',
    },
    [RESOURCE_TYPES.EVENTS]: {
      id: 'RESOURCE_TYPES.EVENTS',
      text: 'Event',
    },
    [SEARCH_RESOURCE_TYPES.GITHUB_ISSUE]: {
      id: 'SEARCH_RESOURCE_TYPES.GITHUB_ISSUE',
      text: 'Github Issue',
    },
    [JOURNEY]: {
      id: 'RESOURCE_TYPES.JOURNEY',
      text: 'Journey',
    },
  },
};
