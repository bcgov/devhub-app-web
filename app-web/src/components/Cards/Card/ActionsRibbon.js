/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import React from 'react';
import PropTypes from 'prop-types';
import { Flag } from 'flag';
import Link from '../../UI/Link/Link';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import Ribbon from './Ribbon';
import shortid from 'shortid';

const ActionsRibbon = ({ actions, repository, owner }) => {
  let actionsRibbon;
  if (!repository || !owner) {
    actionsRibbon = null;
  } else {
    actionsRibbon = actions.map(config => {
      const link = config.getHref(repository, owner);

      return (
        <Flag key={shortid.generate()} name={config.flag}>
          <li>
            <Link to={link} aria-label={config.ariaLabel}>
              <FontAwesomeIcon icon={config.icon} />
            </Link>
          </li>
        </Flag>
      );
    });
  }
  return <Ribbon>{actionsRibbon}</Ribbon>;
};

ActionsRibbon.propTypes = {
  actions: PropTypes.arrayOf(
    PropTypes.shape({
      getHref: PropTypes.func.isRequired,
      icon: PropTypes.node.isRequired,
      ariaLabel: PropTypes.string.isRequired,
      flag: PropTypes.string.isRequired, // this is the dot prop path to the feature flag state reducer
      name: PropTypes.string,
    }).isRequired,
  ),
  repository: PropTypes.string.isRequired,
  owner: PropTypes.string.isRequired,
};

export default ActionsRibbon;
