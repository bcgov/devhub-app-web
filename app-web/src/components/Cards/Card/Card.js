import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faExternalLinkSquareAlt } from '@fortawesome/free-solid-svg-icons';
import styles from './Card.module.css';
import Link from '../../Common/Link';
import { ARIA_LABEL_RESOURCE, ARIA_LABEL_REPO } from '../../../constants/strings';

const Card = ({ sourcePath, sourceName, title, description, resourcePath }) => (
  <article className={styles.Card}>
    <h2>
      <Link to={resourcePath} aria-label={ARIA_LABEL_RESOURCE}>
        <FontAwesomeIcon icon={faExternalLinkSquareAlt} size="1x" />
        <span>{title}</span>
      </Link>
    </h2>
    <div className={styles.Body}>
      <p>{description}</p>
    </div>
    <div className={styles.Actions}>
      {/* <ul>
            </ul> */}
      <Link to={sourcePath} aria-label={ARIA_LABEL_REPO}>
        {sourceName}
      </Link>
    </div>
  </article>
);

Card.displayName = 'Github Card Component';

Card.propTypes = {
  sourcePath: PropTypes.string,
  sourceName: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  resourcePath: PropTypes.string.isRequired,
};

export default Card;
