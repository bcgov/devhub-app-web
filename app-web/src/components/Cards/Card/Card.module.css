.Card {
    width: 250px;
    height: 350px;
    background-color: #fff;
    padding: 14px 17px 64px;
    position: relative;
    color: #3b3b3b;
    display: flex;
    flex-flow: column nowrap;
    flex-shrink: 0;
    margin: 10px 20px 10px 0px;
    transition: box-shadow .25s ease-out;
}

.Card:hover {
    box-shadow: 0 0 2px 2px rgba(0, 44, 95, .25);
}

.Card h2 a {
    font-weight: 400;
    margin-bottom: 13px;
    font-size: 20px;
}

.Card h2 span {
    margin-left: 10px;
    color: #242424;
}

.Card a {
    color: #6d6d6d;
    text-decoration: none;
    font-size: 16px;
    text-transform: capitalize
}

.Card p {
    font-size: 14px;
    line-height: 1.2em;
}
.Body {
    overflow: auto;
    padding-right: 8px;
}
.Actions {
    position: absolute;
    bottom: 0;
    left: 0;
    padding: 8px 14px;
}

.Actions ul {
    list-style: none;
    padding: 0;
    margin: 0;
    display: flex;
    flex-flow: row none;
    justify-content: flex-start;
    align-items: center;
}

.Actions li {
    flex-basis: 25px;
}

@media screen and (min-width: 500px) {
    .Card {
        width: 250px;
        height: 350px;
    }
}
