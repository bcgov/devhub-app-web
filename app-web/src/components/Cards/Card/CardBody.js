/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Derek Siemens
*/

import React from 'react';

import {
  CardDescription,
  CardImage,
  CardImageWrapper,
  CardTitle,
  EventInfoDiv,
  EventDate,
  EventContainer,
  EventImageWrapper,
  MeetupImageWrapper,
} from './index';
import Aux from '../../../hoc/auxillary';
import EventLogo from '../../Event/Logo';

class EventBody extends React.Component {
  render() {
    return (
      <EventContainer>
        <EventDate>
          <span>{this.props.event.start.month}</span>
          {this.props.event.start.day}
          <small>{this.props.event.start.year}</small>
        </EventDate>
        <EventInfoDiv>
          <li>
            {this.props.event.venue !== null ? this.props.event.venue : 'tbd'}
            <this.props.componentType>
              <EventLogo type={this.props.image} />
            </this.props.componentType>
          </li>
        </EventInfoDiv>
      </EventContainer>
    );
  }
}

export class CardBody extends React.Component {
  render() {
    let cardBody = (
      <CardDescription title={this.props.description} clamp={6} tagName="p">
        {this.props.description}
      </CardDescription>
    );

    //Little if statement to change the amount of lines we want to clamp based on if the title takes 1 or two lines
    let clampAmount = 4;
    if (this.props.title.length < 23) {
      clampAmount = 5;
    }

    //first check if its an eventbrite event, as in our resolver in gatsby-node.js -> image is set to "eventbrite"
    if (this.props.image === 'eventbrite' && this.props.description) {
      cardBody = (
        <Aux>
          <CardDescription title={this.props.description} clamp={clampAmount} tagName="p">
            {this.props.event.unfurl.description}
          </CardDescription>
          <EventBody
            event={this.props.event}
            image={this.props.image}
            componentType={EventImageWrapper}
          />
        </Aux>
      );
    } else if (this.props.image === 'meetup' && this.props.description) {
      cardBody = (
        <Aux>
          <CardDescription title={this.props.description} clamp={clampAmount} tagName="p">
            {this.props.event.unfurl.description}
          </CardDescription>
          <EventBody
            event={this.props.event}
            image={this.props.image}
            componentType={MeetupImageWrapper}
          />
        </Aux>
      );
    } else if (this.props.image && this.props.description) {
      cardBody = (
        <Aux>
          <CardDescription title={this.props.description} clamp={2} tagName="p">
            {this.props.description}
          </CardDescription>
          <CardImageWrapper>
            <CardImage src={this.props.image} alt={this.props.title} />
          </CardImageWrapper>
        </Aux>
      );
    } else if (this.props.image) {
      cardBody = (
        <CardImageWrapper>
          <CardImage src={this.props.image} alt={this.props.title} />
        </CardImageWrapper>
      );
    }
    if (this.props.showBody) {
      return (
        <>
          <CardTitle
            clamp={this.props.image && this.props.description ? 2 : 3}
            tagName="h2"
            title={this.props.title}
          >
            {this.props.title}
          </CardTitle>
          {cardBody}
        </>
      );
    } else {
      return <div />;
    }
  }
}
