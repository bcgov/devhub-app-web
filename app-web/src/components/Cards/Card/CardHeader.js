/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faUserCircle,
  faPuzzlePiece,
  faFile,
  faTools,
  faLayerGroup,
  faExternalLinkAlt,
} from '@fortawesome/free-solid-svg-icons';
import { faGithub } from '@fortawesome/fontawesome-free-brands';

import { RESOURCES } from '../../../messages';
import { RESOURCE_TYPES, RESOURCE_TYPES_LIST } from '../../../constants/ui';
// renders the upper ribbon for the card
const FaIcon = styled(FontAwesomeIcon)`
  color: ${props => props.theme.colors[props.type]};
`;

const H3 = styled.h3`
  color: #444;
  font-size: 14px;
  margin-bottom: 0;
  font-weight: 400;
`;

const CardHeader = ({ type, linksToExternal }) => {
  let icon = null;

  // eslint-disable-next-line default-case
  switch (type) {
    case RESOURCE_TYPES.COMPONENTS:
      icon = faPuzzlePiece;
      break;
    case RESOURCE_TYPES.DOCUMENTATION:
      icon = faFile;
      break;
    case RESOURCE_TYPES.PEOPLE:
      icon = faUserCircle;
      break;
    case RESOURCE_TYPES.SELF_SERVICE_TOOLS:
      icon = faTools;
      break;
    case RESOURCE_TYPES.COLLECTIONS:
      icon = faLayerGroup;
      break;
    case RESOURCE_TYPES.REPOSITORIES:
      icon = faGithub;
      break;
  }

  return (
    <div
      css={css`
        margin-bottom: 4px;
      `}
    >
      <H3>
        <FaIcon type={type} icon={icon} />{' '}
        <span
          css={css`
            margin: 4px;
          `}
        >
          {RESOURCES.types[type].defaultMessage}
        </span>
        {linksToExternal && (
          <small>
            <FontAwesomeIcon icon={faExternalLinkAlt} />
          </small>
        )}
      </H3>
    </div>
  );
};

CardHeader.propTypes = {
  type: PropTypes.oneOf(RESOURCE_TYPES_LIST),
  linksToExternal: PropTypes.bool.isRequired,
};

CardHeader.defaultProps = {
  linksToExternal: false,
};

export default CardHeader;
