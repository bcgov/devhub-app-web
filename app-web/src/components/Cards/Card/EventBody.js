/*
Copyright 2019 Province of British Columbia
Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 
   http://www.apache.org/licenses/LICENSE-2.0
Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Derek Siemens
*/

import React from 'react';
import PropTypes from 'prop-types';

import { EventInfoDiv, EventDate, EventContainer } from './index';
import EventLogo from '../../Event/Logo';

class EventBody extends React.Component {
  render() {
    return (
      <EventContainer>
        <EventDate>
          <span>{this.props.event.start.month}</span>
          {this.props.event.start.day}
          <small>{this.props.event.start.year}</small>
        </EventDate>
        <EventInfoDiv>
          <li>
            {this.props.event.venue !== null ? this.props.event.venue : 'tbd'}
            <this.props.componentType>
              <EventLogo type={this.props.image} />
            </this.props.componentType>
          </li>
        </EventInfoDiv>
      </EventContainer>
    );
  }
}

EventBody.propTypes = {
  event: PropTypes.object.isRequired,
};

export default EventBody;
