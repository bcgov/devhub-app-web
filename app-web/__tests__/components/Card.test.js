import React from 'react';
import { shallow } from 'enzyme';
import Card from '../../src/components/Cards/Card/Card';

describe('Card Component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(
      <Card
        title="title"
        description="description"
        resourcePath="resourcePath"
        sourcePath="sourcePath"
        sourceName="sourceName"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
