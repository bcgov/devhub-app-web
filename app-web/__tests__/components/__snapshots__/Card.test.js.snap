// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Github Card Component
    description="description"
    resourcePath="resourcePath"
    sourceName="sourceName"
    sourcePath="sourcePath"
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          <GatsbyLink
            aria-label="Learn more about this article"
            to="resourcePath"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f360",
                    "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                  ],
                  "iconName": "external-link-square-alt",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size="1x"
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <span>
              title
            </span>
          </GatsbyLink>
        </h2>,
        <div
          className="Body"
        >
          <p>
            description
          </p>
        </div>,
        <div
          className="Actions"
        >
          <GatsbyLink
            aria-label="Link to the Github Repository"
            to="sourcePath"
          >
            sourceName
          </GatsbyLink>
        </div>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <GatsbyLink
            aria-label="Learn more about this article"
            to="resourcePath"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f360",
                    "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                  ],
                  "iconName": "external-link-square-alt",
                  "prefix": "fas",
                }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size="1x"
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
            <span>
              title
            </span>
          </GatsbyLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "aria-label": "Learn more about this article",
            "children": Array [
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f360",
                      "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                    ],
                    "iconName": "external-link-square-alt",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="1x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />,
              <span>
                title
              </span>,
            ],
            "to": "resourcePath",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f360",
                    "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                  ],
                  "iconName": "external-link-square-alt",
                  "prefix": "fas",
                },
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": "1x",
                "spin": false,
                "symbol": false,
                "title": "",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "title",
              },
              "ref": null,
              "rendered": "title",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            description
          </p>,
          "className": "Body",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "description",
          },
          "ref": null,
          "rendered": "description",
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <GatsbyLink
            aria-label="Link to the Github Repository"
            to="sourcePath"
          >
            sourceName
          </GatsbyLink>,
          "className": "Actions",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "aria-label": "Link to the Github Repository",
            "children": "sourceName",
            "to": "sourcePath",
          },
          "ref": null,
          "rendered": "sourceName",
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            <GatsbyLink
              aria-label="Learn more about this article"
              to="resourcePath"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f360",
                      "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                    ],
                    "iconName": "external-link-square-alt",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="1x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              <span>
                title
              </span>
            </GatsbyLink>
          </h2>,
          <div
            className="Body"
          >
            <p>
              description
            </p>
          </div>,
          <div
            className="Actions"
          >
            <GatsbyLink
              aria-label="Link to the Github Repository"
              to="sourcePath"
            >
              sourceName
            </GatsbyLink>
          </div>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <GatsbyLink
              aria-label="Learn more about this article"
              to="resourcePath"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f360",
                      "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                    ],
                    "iconName": "external-link-square-alt",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size="1x"
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
              <span>
                title
              </span>
            </GatsbyLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "aria-label": "Learn more about this article",
              "children": Array [
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        448,
                        512,
                        Array [],
                        "f360",
                        "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                      ],
                      "iconName": "external-link-square-alt",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size="1x"
                  spin={false}
                  symbol={false}
                  title=""
                  transform={null}
                />,
                <span>
                  title
                </span>,
              ],
              "to": "resourcePath",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f360",
                      "M448 80v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48h352c26.51 0 48 21.49 48 48zm-88 16H248.029c-21.313 0-32.08 25.861-16.971 40.971l31.984 31.987L67.515 364.485c-4.686 4.686-4.686 12.284 0 16.971l31.029 31.029c4.687 4.686 12.285 4.686 16.971 0l195.526-195.526 31.988 31.991C358.058 263.977 384 253.425 384 231.979V120c0-13.255-10.745-24-24-24z",
                    ],
                    "iconName": "external-link-square-alt",
                    "prefix": "fas",
                  },
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": "1x",
                  "spin": false,
                  "symbol": false,
                  "title": "",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "title",
                },
                "ref": null,
                "rendered": "title",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              description
            </p>,
            "className": "Body",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "description",
            },
            "ref": null,
            "rendered": "description",
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <GatsbyLink
              aria-label="Link to the Github Repository"
              to="sourcePath"
            >
              sourceName
            </GatsbyLink>,
            "className": "Actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "aria-label": "Link to the Github Repository",
              "children": "sourceName",
              "to": "sourcePath",
            },
            "ref": null,
            "rendered": "sourceName",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
